#13 Windows Installer XML project to create msi installer

drmason789 opened the pull request.

drmason789 committed:
#10 Optimized build configurations

drmason789 committed:
#13 Windows Installer XML project to create msi installer

leibnitz27 commented on olewoo_installer/olewoo_installer.wixproj:
```diff
@@ -0,0 +1,46 @@
+  </ItemGroup>
+  <ItemGroup>
+    <WixExtension Include="WixNetFxExtension">
+      <HintPath>..\..\..\Program Files (x86)\WiX Toolset v3.11\bin\WixNetFxExtension.dll</HintPath>
```
Having a hardcoded path like this doesn't seem right.  Is this the standard way to build these?  Is this available from an environment variable?

drmason789 commented:
I agree this is not ideal, but unfortunately the WIX Toolset installer doesn't put it in the GAC. 
I see there's actually a nuget package for WIX now, so I can use a relative reference to the packages folder. I'll do that.

drmason789 committed:
#13 Use nuget package for WiX instead and reference the local packages DLL instead of the one in Program Files (x86) which would be different in 32-bit Windows.

drmason789 commented:
There's a [Visual Studio extension](https://wixtoolset.org/releases/) available that integrates WIX with VS.

drmason789 committed:
#13 made use of the olewoo icon in the start menu, and the app list/programs and features list. Put the olewoo shortcut at the top level of the start menu instead of inside a folder.

# Please enter the commit message for your changes. Lines starting
# with '#' will be kept; you may remove them yourself if you want to.
# An empty message aborts the commit.
#
# Date:      Mon May 25 15:16:31 2020 +0100
#
# On branch WIX_Installer
# Your branch is up to date with 'origin/WIX_Installer'.
#
# Changes to be committed:
#	modified:   olewoo_installer/Setup.wxs
#

leibnitz27 commented on olewoo_installer/Setup.wxs:
```diff
@@ -0,0 +1,107 @@
+  <?define SOURCE="..\\$(var.Configuration)"?>
+  <?define PRODUCT_NAME = "OLEWOO"?>
+  <?define PRODUCT_DESC = "Type Library Viewer"?>
+  <?define MANUFACTURER="Liebnitz"?>
```
L_ei_bnitz after Gottfried Wilhelm, ;)   But might be easier if you just put github url as manufacturer?
I know it's not quite canon, but works.

Is there any reason it's all in capitals?  I don't know about you but I find caps really annoying in application names/paths.  (Not a huge deal, but while I'm here!)

drmason789 replied:
Personally, I don't like pascal cased acronyms .e.g Ole, Sql, Xml. I don't even know what "woo" is, other than something that men did to women in days of yore.

But it's your project. I'll change it to OleWoo like you have in the title bar of the UI.

I'll just make it the way you prefer. Would you like a license acceptance screen (I'd use the first pre element from license.txt? This could also lead to choosing background and banner images.

leibnitz27 replied:
'Woo'? Honestly, it was just the first vaguely homophonic thing that sprang to mind when slavishly reimplementing *cough* oleview.   There's also a vague connotation of success.   I was not cut out for marketing.

Casing - TBH I naturally favour all lower, as I spend 90% of my time in unixland, but I'm happy with CaMel or Initialcaps, as those seem to be the Windows standards. (from a brief browse of my progra~1).

Licence - Genuinely not bothered.  ;)

drmason789 replied:
I found out how to do the manufacturerUrl  - there's a property you have to set. This makes no difference to Settings/Apps but it does to Programs and Features:

![image](https://user-images.githubusercontent.com/29954900/82834062-e2d47680-9eb7-11ea-982a-e9fece399601.png)

drmason789 committed:
#13 Changed OLEWOO to OleWoo. Changed Manufacturer to github URL for OleWoo. Changed some UK spelling to US (licence to license).

drmason789 committed:
#13 Manufacturer links and comments

drmason789 committed:
#13 commented out superfluous link

drmason789 committed:
Merged changes for #10

drmason789 commented:
I've merged the changes for #10. 
I'm content to consider this finished if you are. 

To do releases, you need to update the version number in the version.wxi file. I have it in a separate file so it's easier to script the update.
